src/overlays/actors/ovl_Bg_Spot16_Bombstone/z_bg_spot16_bombstone.c

```diff
@@ -21,49 +111,445 @@ const ActorInit Bg_Spot16_Bombstone_InitVars = {
+        CollisionCheck_SetOC(globalCtx, &globalCtx->colChkCtx, &this->colliderJntSph);
+        CollisionCheck_SetAC(globalCtx, &globalCtx->colChkCtx, &this->colliderJntSph);
+    }
+
+    if (mREG(64) == 1) {
+        func_808B561C(this, globalCtx);
+        mREG(64) = -10;
+    } else if (mREG(64) < 0) {
+        mREG(64)++;
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Spot16_Bombstone/func_808B4D04.s")
+void func_808B5A78(BgSpot16Bombstone* this) {
+    this->unk_154 = 0;
+    this->unk_158 = 0;
+    this->actor.draw = NULL;
+    this->actionFunc = func_808B5A94;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Spot16_Bombstone/func_808B4D9C.s")
+void func_808B5A94(BgSpot16Bombstone* this, GlobalContext* globalCtx) {
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Spot16_Bombstone/func_808B4E58.s")
+    func_808B5240(this, globalCtx);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Spot16_Bombstone/BgSpot16Bombstone_Init.s")
+    if (this->unk_154 == 0x38) {
+        func_80078884(NA_SE_SY_CORRECT_CHIME);
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Spot16_Bombstone/BgSpot16Bombstone_Destroy.s")
+    if (this->unk_154 >= 0x3D) {
+        Actor_Kill(&this->actor);
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Spot16_Bombstone/func_808B51A8.s")
+void func_808B5AF0(BgSpot16Bombstone* this) {
+    this->actionFunc = func_808B5B04;
+    this->actor.draw = NULL;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Spot16_Bombstone/func_808B5240.s")
+void func_808B5B04(BgSpot16Bombstone* this, GlobalContext* globalCtx) {
+    if (Object_IsLoaded(&globalCtx->objectCtx, this->bombiwaBankIndex)) {
+        func_808B5B58(this);
+        this->actor.draw = BgSpot16Bombstone_Draw;
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Spot16_Bombstone/func_808B53A8.s")
+void func_808B5B58(BgSpot16Bombstone* this) {
+    this->unk_154 = 0;
+    this->actionFunc = func_808B5B6C;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Spot16_Bombstone/func_808B561C.s")
+void func_808B5B6C(BgSpot16Bombstone* this, GlobalContext* globalCtx) {
+    u16 bgCheckFlags;
+    Actor* actor = &this->actor;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Spot16_Bombstone/func_808B56BC.s")
+    Actor_MoveForward(actor);
+    actor->shape.rot.x = actor->shape.rot.x + this->unk_210;
+    actor->shape.rot.z = actor->shape.rot.z + this->unk_212;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Spot16_Bombstone/func_808B57E0.s")
+    if (this->unk_154 >= 0x3D) {
+        Actor_Kill(actor);
+        return;
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Spot16_Bombstone/func_808B5934.s")
+    bgCheckFlags = actor->bgCheckFlags;
+    if (bgCheckFlags & 8 || (bgCheckFlags & 1 && actor->velocity.y < 0.0f)) {
```

Comment:
Any chance this works to get rid of the temp?
```suggestion
    if (actor->bgCheckFlags & 8 || (actor->bgCheckFlags & 1 && actor->velocity.y < 0.0f)) {
```